### PR TITLE
feat(copy): append original description

### PR DESCRIPTION
**Pull request by v1v**
## Description of the change

Append original description to the copy and backport commands

## Related issues

Fix https://github.com/Mergifyio/mergify-engine/issues/2414

### Development

- [ ] All checks must pass (Semantic Pull Request, pep8, requirements, unit tests, functional tests, security checks, …)
- [ ] The code changed/added as part of this pull request must be covered with tests
- [ ] Hotfixes must have a link to Sentry issue and the ``hotfix`` label
- [ ] Features must have a link to a Linear task

### Code Review

Code review policies are handled and automated by Mergify.

* When all tests pass, reviewers will be assigned automatically.
* When change is approved by at least one review and no pending review are
  remaining, pull request is retested against its base branch.
* The pull request is then merged automatically.

### CODE REVIEW

File: mergify_engine/duplicate_pull.py [hunk 1]
```diff
@@ -319,6 +321,9 @@ async def duplicate(
             "checking-out-pull-requests-locally/"
         )
 
+    if append_original_description and not ctxt.pull["body"] is None:
```
Comment (v1v) [inline mergify_engine/duplicate_pull.py:324]: I quickly submitted my draft PR but it's failing when running the flake8 with

```
TypedDict "GitHubPullRequest" has no key 'body'  [misc]
```

I somehow though the `body` attribute was a correct one. I'll need to debug further

File: mergify_engine/actions/copy.py [hunk 2]
```diff
@@ -55,6 +55,7 @@ def get_config_schema(
             voluptuous.Required("branches", default=[]): [str],
             voluptuous.Required("regexes", default=[]): [voluptuous.Coerce(Regex)],
             voluptuous.Required("ignore_conflicts", default=True): bool,
+            voluptuous.Required("append_original_description", default=False): bool,
```
Comment (sileht) [inline mergify_engine/actions/copy.py:58]: I got the idea, but to make it more generic like the title, maybe the new attribute should just be:

```
voluptuous.Required("body_suffix", default=""): types.Jinja2)
```

And the configuration will look like:

```
backport:
    body_suffix: "{{ body }}"
```

WDYT @jd ?

Comment (jd) [reply mergify_engine/actions/copy.py:58]: What about having the whole body being customizable?
Where you could reuse part of the original pull request like `{{ body }}` in it and maybe some other data such as `{{ git_cherry_pick_output }}` etc?

File: mergify_engine/duplicate_pull.py (hunk 1 above)
Comment (jd) [reply mergify_engine/duplicate_pull.py:324]: `body` is valid but it's just missing from the type definition in `github_types`.
Those definitions are not always complete (we're lazy). You can double-check GitHub API if you doubt.

File: mergify_engine/actions/copy.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/actions/copy.py:58]: We can do that too, `rendering_template()` already support additional variables

**Comment by sileht** [comment]
Replaced by https://github.com/Mergifyio/mergify-engine/pull/2479

**Pull request closed without merging**